### PR TITLE
Replace Locks in AmqpEventHubClient and Code Clean ups in AmqpEventHubClient

### PR DESCRIPTION
**Pull request by davidrevoledo**
## Description
- Acording to this is not really safe the double lock pattern w/o volatile at the bottom of the link is the official reference, here  is the change to replace specifically in `AmqpEventHubClient` for a Lazy instead.

Reference : https://help.semmle.com/wiki/display/CSHARP/Double-checked+lock+is+not+thread-safe

**AmqpEventHubClient Code CleanUps :** 
 - Remove internal to a few static methods that are only using privately  this will help if we can expose an interface for testing purpose

-  In OnGetPartitionRuntimeInformationAsync / OnGetRuntimeInformationAsync remove await and return task instead

- Create common Constructor to avoid repeated code in both ctors

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](./CONTRIBUTING.md).**
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR.
- [x] The pull request does not introduce breaking changes (unless a major version change occurs in the assembly and module).
- [x] If applicable, the public code is properly documented.
- [x] Pull request includes test coverage for the included changes.
- [x] The code builds without any errors.

### CODE REVIEW

File: src/Microsoft.Azure.EventHubs/EventHubClient.cs [hunk 1]
```diff
@@ -599,10 +580,9 @@ internal EventDataSender CreateEventSender(string partitionId = null)
         /// </summary>
         protected override void OnRetryPolicyUpdate()
         {
-            // Propagate retry policy updates to inner sender if there is any.
-            if (this.innerSender != null)
+            if (this.innerSender.IsValueCreated)
```
Comment (serkantkaraca) [inline src/Microsoft.Azure.EventHubs/EventHubClient.cs]: >this.innerSender.IsValueCreated [](start = 16, length = 31)

This defeats the lazy purpose and creates innerSender. innserSender should only be created for 'Send' purpose only.

Comment (serkantkaraca) [reply src/Microsoft.Azure.EventHubs/EventHubClient.cs]: You are so right. I seemed to have missed lazy initialization note.

---
In reply to: [232746929](https://github.com/Azure/azure-event-hubs-dotnet/pull/345#discussion_r232746929) [](ancestors = 232746929)

Comment (serkantkaraca) [inline src/Microsoft.Azure.EventHubs/EventHubClient.cs]: >innerSender [](start = 21, length = 11)

Can you add null check? Tests are failing with NullRef here.

Comment (davidrevoledo) [reply src/Microsoft.Azure.EventHubs/EventHubClient.cs]: ok

**Pull request merged**